src/Lumi/Components/Loader.purs

```diff
@@ -24,29 +26,103 @@ loader = makeStateless component $ loaderElement <<< mapProps
+          , "&[data-color=\"accent-3\"]": loaderColorMixin colors.accent3
+          , "&[data-color=\"accent-33\"]": loaderColorMixin colors.accent33
+          -- bg color (needed for fading loader center)
+          , "&[data-bg-color=\"primary\"]::after": loaderBgColorMixin colors.primary
+          , "&[data-bg-color=\"primary-1\"]::after": loaderBgColorMixin colors.primary1
+          , "&[data-bg-color=\"primary-2\"]::after": loaderBgColorMixin colors.primary2
+          , "&[data-bg-color=\"primary-3\"]::after": loaderBgColorMixin colors.primary3
+          , "&[data-bg-color=\"primary-4\"]::after": loaderBgColorMixin colors.primary4
+          , "&[data-bg-color=\"secondary\"]::after": loaderBgColorMixin colors.secondary
+          , "&[data-bg-color=\"white\"]::after": loaderBgColorMixin colors.white
+          , "&[data-bg-color=\"black\"]::after": loaderBgColorMixin colors.black
+          , "&[data-bg-color=\"black-1\"]::after": loaderBgColorMixin colors.black1
+          , "&[data-bg-color=\"black-2\"]::after": loaderBgColorMixin colors.black2
+          , "&[data-bg-color=\"black-3\"]::after": loaderBgColorMixin colors.black3
+          , "&[data-bg-color=\"black-4\"]::after": loaderBgColorMixin colors.black4
+          , "&[data-bg-color=\"black-5\"]::after": loaderBgColorMixin colors.black5
+          , "&[data-bg-color=\"black-6\"]::after": loaderBgColorMixin colors.black6
+          , "&[data-bg-color=\"black-7\"]::after": loaderBgColorMixin colors.black7
+          , "&[data-bg-color=\"black-8\"]::after": loaderBgColorMixin colors.black8
+          , "&[data-bg-color=\"accent-1\"]::after": loaderBgColorMixin colors.accent1
+          , "&[data-bg-color=\"accent-2\"]::after": loaderBgColorMixin colors.accent2
+          , "&[data-bg-color=\"accent-3\"]::after": loaderBgColorMixin colors.accent3
+          , "&[data-bg-color=\"accent-33\"]::after": loaderBgColorMixin colors.accent33
+          }
       , "@keyframes spin":
           { from: { transform: "rotate(0deg)" }
           , to: { transform: "rotate(360deg)" }
           }
       }
   }
+  where
+    loaderColorMixin value = spinnerMixin
+      { radius: "3.8rem"
+      , borderWidth: "0.5rem"
+      , color: cssStringHSLA value
+      }
 
-spinnerMixin :: { radius :: String, borderWidth :: String } -> JSS
-spinnerMixin { radius, borderWidth } = jss
-  { boxSizing: "border-box"
-  , content: "\"\""
-  , display: "inline-block"
+    loaderBgColorMixin value =
+      { background: cssStringHSLA value
+      }
+
+spinnerMixin :: { radius :: String, borderWidth :: String, color :: String } -> JSS
+spinnerMixin { radius, borderWidth, color } = jss
+  { width: radius
   , height: radius
-  , width: radius
-  , border: [ borderWidth, "solid", cssStringHSLA colors.black1 ]
-  , borderTopColor: cssStringHSLA colors.black4
   , borderRadius: "50%"
+  , background: "linear-gradient(to right, " <> color <> " 10%, rgba(255, 255, 255, 0) 42%)"
```

Comment:
Shouldn't we just fade from `color` with 100% opacity to `color` with 0% opacity? I think this would remove the need for specifying a background color to the loader.

Reply:
Did you by chance get to try this out @kimispencer ?

Reply:
Perhaps I am misunderstanding your comment, but the new design has a hard edge, and then fades at the end of the circle... I wasn't able to accomplish this one-directional fade on the border without using a `linear-gradient` and thus needed background. Let me know if you meant something else.

Reply:
Oh, sorry, I think I didn't explain myself properly. I thought of using border gradients for that, but it's just not possible. My main concern with this is that, since the background is a solid color, we won't be able to use loading spinners over text, otherwise this would happen:
![Screen Shot 2019-08-06 at 10 13 21](https://user-images.githubusercontent.com/2164548/62542844-fe8c6480-b832-11e9-9b07-ac671494383c.png)

Maybe we should use baked images or SVG for this, since we have a closed set of possible colors?

I don't think we'll be using loaders in all colors we offer, perhaps we should just stick to 2 or 3 colors? Would make this easier if we go down the image/SVG path.

Reply:
Yeah I was trying to use a CSS-only approach but it does result in the spinner not having an opaque/transparent center... I see your point that this might not look good over text. I think rather than a baked in image it would be more flexible (+ look cleaner as vector) using SVG. I'll implement a version using that approach 👍